### DISCUSSION TITLE
Hosting on a subpath

### DISCUSSION THREAD

**Discussion by answerquest** [minio/minio#16237] [Q&A] [answered]
I have an Apache setup with SSL on a webserver. I've setup proxy/reverseproxy to forward the minio ports to subpaths:

    ProxyPass /datastore1/ http://127.0.0.1:9011/
    ProxyPassReverse /datastore1/ http://127.0.0.1:9011/

    ProxyPass /datastore2/ http://127.0.0.1:9012/
    ProxyPassReverse /datastore2/ http://127.0.0.1:9012/

Running by docker:
```
docker run \
   --rm -it \
   -p 9011:9000 \
   -p 9012:9090 \
   --user $(id -u):$(id -g) \
   --name minio1 \
   -e "MINIO_ROOT_USER=user1" \
   -e "MINIO_ROOT_PASSWORD=password1" \
   -v $(pwd)/minio:/data \
   -e "MINIO_BROWSER_REDIRECT_URL=https://website.com/datastore2/" \
   quay.io/minio/minio:latest server /data --console-address ":9090"
```

The UI console runs fine on https://website.com/datastore2/ .

In python program, if I try to set the endpoint to "website.com/datastore1", getting this:  
`ValueError: path in endpoint is not allowed`

It's going to be a long-winded process to setup a new subdomain for minio. I just want to use this for development, so that I can connect to the on-server instance from my local system securely. When in production it's going to be on the same server as the program and so localhost:9011 will suffice.

 This is working if I `ufw allow 9011` and :
````
client = Minio(
    "website.com:9011",
    access_key="something",
    secret_key="somekey",
    secure=False
)
```

But that exposes the credentials to a non-secure way.  
What should I do here?

**Comment by harshavardhana** [comment, marked as answer]
S3 API cannot be hosted on subpath, of course UI will run fine.

**Reply by siddjellali** [reply]
Hello, 

I noticed it's work with nginx but not on traefik. 
Any idea why ? 

```
<Error>
<Code>SignatureDoesNotMatch</Code>
<Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message>
<Key>VSR-SLS.xlsx</Key>
<BucketName>auditor</BucketName>
<Resource>/auditor/VSR-SLS.xlsx</Resource>
<RequestId>17C1A77BDD4708B0</RequestId>
<HostId>dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8</HostId>
</Error>
```

**Reply by henri9813** [reply]
Hello, 

how did you make it works using nginx ?

**Comment by landsman** [comment]
Did someone figure out how to host on subpath? Sometimes it would be really useful.

**Reply by prakashsvmx** [reply]
If it is related to  UI, please refer to https://github.com/minio/object-browser/tree/master/web-app/tests/subpath-nginx
